Project builds fail after esp-idf update/clean install of latest

### The issue

After updating my esp-idf installation, I observed build fails for projects. I reinstalled esp-idf entirely, the build issue persists. 

ESP-IDF commit:

commit 31b7694551620522cc3fde0623321a38bfea762f (HEAD -> master, origin/master, origin/HEAD)
Merge: 0a511e576e c063cbd7f7
Author: Roland Dobai <roland@espressif.com>
Date:   Mon May 16 21:50:16 2022 +0800

    Merge branch 'contrib/github_pr_8955' into 'master'

ESP-IDF-LIB commit:

commit af5f7ad3d3d80d7bb33b1293a973727daff6d52e (HEAD -> master, origin/master, origin/HEAD)
Author: UncleRus <unclerus@gmail.com>
Date:   Wed May 11 02:22:18 2022 +0500

    i2cdev: add operation type parameter to i2c_dev_probe()

Build error when building esp-idf-lib/examples/bmp280:

In file included from /home/eldritchjs/esp/esp-idf-lib/components/lib8tion/lib8tion.c:24:
/home/eldritchjs/esp/esp-idf-lib/components/lib8tion/lib8tion.h:178:10: fatal error: esp_timer.h: No such file or directory
 #include <esp_timer.h>
          ^~~~~~~~~~~~~
compilation terminated.


esp_timer.h still exists within my ESP-IDF path, Currently looking through includes variables to see what's changed to cause the project build to fail


### Which SDK are you using?

esp-idf

### Which version of SDK are you using?

master

### Which build target have you used?

- [X] esp32
- [ ] esp32s2
- [ ] esp32s3
- [ ] esp32c2
- [ ] esp8266
- [X] other

### Component causing the issue

lib8tion

### Anything in the logs that might be useful for us?

```console
In file included from /home/eldritchjs/esp/esp-idf-lib/components/lib8tion/lib8tion.c:24:
/home/eldritchjs/esp/esp-idf-lib/components/lib8tion/lib8tion.h:178:10: fatal error: esp_timer.h: No such file or directory
 #include <esp_timer.h>
          ^~~~~~~~~~~~~
compilation terminated.
```


### Additional information or context

I've attempted the build with targets esp32 and esp32c3

### Confirmation

- [X] This report is not a question nor a request for drivers.

--- a/components/button/CMakeLists.txt
+++ b/components/button/CMakeLists.txt
@@ -1,6 +1,8 @@
 if(${IDF_TARGET} STREQUAL esp8266)
     set(req esp8266 esp_timer)
-else()
+elseif(${IDF_VERSION_MAJOR} STREQUAL 4 AND ${IDF_VERSION_MINOR} STREQUAL 1 AND ${IDF_VERSION_PATCH} STREQUAL 3)
+    set(req driver)
+else()    
     set(req driver esp_timer)
 endif()
 

--- a/components/button/CMakeLists.txt
+++ b/components/button/CMakeLists.txt
@@ -1,7 +1,7 @@
 if(${IDF_TARGET} STREQUAL esp8266)
-    set(req esp8266)
+    set(req esp8266 esp_timer)
 else()
-    set(req driver)
+    set(req driver esp_timer)
 endif()
 
 idf_component_register(

--- a/components/color/CMakeLists.txt
+++ b/components/color/CMakeLists.txt
@@ -1,5 +1,13 @@
+if(${IDF_VERSION_MAJOR} STREQUAL 4 AND ${IDF_VERSION_MINOR} STREQUAL 1 AND ${IDF_VERSION_PATCH} STREQUAL 3)
+	message("Not including esp_timer in color")
+    	set(req lib8tion)	
+else()
+    message("Including esp_timer in color")
+    set(req lib8tion)	
+endif()
+
 idf_component_register(
     SRCS color.c
     INCLUDE_DIRS .
-    REQUIRES lib8tion esp_timer
+    REQUIRES ${req}
 )

--- a/components/color/CMakeLists.txt
+++ b/components/color/CMakeLists.txt
@@ -1,5 +1,5 @@
 idf_component_register(
     SRCS color.c
     INCLUDE_DIRS .
-    REQUIRES lib8tion
+    REQUIRES lib8tion esp_timer
 )

--- a/components/encoder/CMakeLists.txt
+++ b/components/encoder/CMakeLists.txt
@@ -1,5 +1,7 @@
 if(${IDF_TARGET} STREQUAL esp8266)
     set(req esp8266 freertos log esp_timer)
+elseif(${IDF_VERSION_MAJOR} STREQUAL 4 AND ${IDF_VERSION_MINOR} STREQUAL 1 AND ${IDF_VERSION_PATCH} STREQUAL 3)
+    set(req driver freertos log)
 else()
     set(req driver freertos log esp_timer)
 endif()

--- a/components/encoder/CMakeLists.txt
+++ b/components/encoder/CMakeLists.txt
@@ -1,7 +1,7 @@
 if(${IDF_TARGET} STREQUAL esp8266)
-    set(req esp8266 freertos log)
+    set(req esp8266 freertos log esp_timer)
 else()
-    set(req driver freertos log)
+    set(req driver freertos log esp_timer)
 endif()
 
 idf_component_register(

--- a/components/hdc1000/CMakeLists.txt
+++ b/components/hdc1000/CMakeLists.txt
@@ -1,5 +1,5 @@
 idf_component_register(
     SRCS hdc1000.c
     INCLUDE_DIRS .
-    REQUIRES i2cdev log esp_idf_lib_helpers
+    REQUIRES i2cdev log esp_idf_lib_helpers esp_timer
 )

--- a/components/hdc1000/CMakeLists.txt
+++ b/components/hdc1000/CMakeLists.txt
@@ -1,5 +1,11 @@
+if(${IDF_VERSION_MAJOR} STREQUAL 4 AND ${IDF_VERSION_MINOR} STREQUAL 1 AND ${IDF_VERSION_PATCH} STREQUAL 3)
+	set(req i2cdev log esp_idf_lib_helpers)
+else()
+	set(req i2cdev log esp_idf_lib_helpers esp_timer)
+endif()
+
 idf_component_register(
     SRCS hdc1000.c
     INCLUDE_DIRS .
-    REQUIRES i2cdev log esp_idf_lib_helpers esp_timer
+    REQUIRES ${req}
 )

--- a/components/hmc5883l/CMakeLists.txt
+++ b/components/hmc5883l/CMakeLists.txt
@@ -1,5 +1,11 @@
+if(${IDF_VERSION_MAJOR} STREQUAL 4 AND ${IDF_VERSION_MINOR} STREQUAL 1 AND ${IDF_VERSION_PATCH} STREQUAL 3)
+	set(req i2cdev log esp_idf_lib_helpers)
+else()
+	set(req i2cdev log esp_idf_lib_helpers esp_timer)
+endif()
+
 idf_component_register(
     SRCS hmc5883l.c
     INCLUDE_DIRS .
-    REQUIRES i2cdev log esp_idf_lib_helpers esp_timer
+    REQUIRES ${req}
 )

--- a/components/hmc5883l/CMakeLists.txt
+++ b/components/hmc5883l/CMakeLists.txt
@@ -1,5 +1,5 @@
 idf_component_register(
     SRCS hmc5883l.c
     INCLUDE_DIRS .
-    REQUIRES i2cdev log esp_idf_lib_helpers
+    REQUIRES i2cdev log esp_idf_lib_helpers esp_timer
 )

--- a/components/hts221/CMakeLists.txt
+++ b/components/hts221/CMakeLists.txt
@@ -1,5 +1,11 @@
+if(${IDF_VERSION_MAJOR} STREQUAL 4 AND ${IDF_VERSION_MINOR} STREQUAL 1 AND ${IDF_VERSION_PATCH} STREQUAL 3)
+	set(req i2cdev log esp_idf_lib_helpers)
+else()
+	set(req i2cdev log esp_idf_lib_helpers esp_timer)
+endif()
+
 idf_component_register(
     SRCS hts221.c
     INCLUDE_DIRS .
-    REQUIRES i2cdev log esp_idf_lib_helpers esp_timer
+    REQUIRES ${req}
 )

--- a/components/hts221/CMakeLists.txt
+++ b/components/hts221/CMakeLists.txt
@@ -1,5 +1,5 @@
 idf_component_register(
     SRCS hts221.c
     INCLUDE_DIRS .
-    REQUIRES i2cdev log esp_idf_lib_helpers
+    REQUIRES i2cdev log esp_idf_lib_helpers esp_timer
 )

--- a/components/hx711/CMakeLists.txt
+++ b/components/hx711/CMakeLists.txt
@@ -1,7 +1,7 @@
 if(${IDF_TARGET} STREQUAL esp8266)
-    set(req esp8266 freertos esp_idf_lib_helpers)
+    set(req esp8266 freertos esp_idf_lib_helpers esp_timer)
 else()
-    set(req driver freertos esp_idf_lib_helpers)
+    set(req driver freertos esp_idf_lib_helpers esp_timer)
 endif()
 
 idf_component_register(

--- a/components/hx711/CMakeLists.txt
+++ b/components/hx711/CMakeLists.txt
@@ -1,5 +1,7 @@
 if(${IDF_TARGET} STREQUAL esp8266)
     set(req esp8266 freertos esp_idf_lib_helpers esp_timer)
+elseif(${IDF_VERSION_MAJOR} STREQUAL 4 AND ${IDF_VERSION_MINOR} STREQUAL 1 AND ${IDF_VERSION_PATCH} STREQUAL 3)
+    	set(req driver freertos esp_idf_lib_helpers)
 else()
     set(req driver freertos esp_idf_lib_helpers esp_timer)
 endif()

--- a/components/lib8tion/CMakeLists.txt
+++ b/components/lib8tion/CMakeLists.txt
@@ -1,4 +1,5 @@
 idf_component_register(
     INCLUDE_DIRS .
     SRCS lib8tion.c
+    REQUIRES esp_timer
 )

--- a/components/lib8tion/CMakeLists.txt
+++ b/components/lib8tion/CMakeLists.txt
@@ -1,5 +1,11 @@
+if(${IDF_VERSION_MAJOR} STREQUAL 4 AND ${IDF_VERSION_MINOR} STREQUAL 1 AND ${IDF_VERSION_PATCH} STREQUAL 3)
+	set(req)	
+else()
+    set(req esp_timer)
+endif()
+
 idf_component_register(
     INCLUDE_DIRS .
     SRCS lib8tion.c
-    REQUIRES esp_timer
+    REQUIRES ${req}
 )

--- a/components/mhz19b/CMakeLists.txt
+++ b/components/mhz19b/CMakeLists.txt
@@ -1,3 +1,9 @@
+if(${IDF_VERSION_MAJOR} STREQUAL 4 AND ${IDF_VERSION_MINOR} STREQUAL 1 AND ${IDF_VERSION_PATCH} STREQUAL 3)
+	set(req log esp_idf_lib_helpers driver)
+else()
+	set(req log esp_idf_lib_helpers driver esp_timer)
+endif()
+
 idf_component_register(SRCS "mhz19b.c"
                        INCLUDE_DIRS "."
-                       REQUIRES log esp_idf_lib_helpers driver esp_timer)
+                       REQUIRES ${req})

--- a/components/mhz19b/CMakeLists.txt
+++ b/components/mhz19b/CMakeLists.txt
@@ -1,3 +1,3 @@
 idf_component_register(SRCS "mhz19b.c"
                        INCLUDE_DIRS "."
-                       REQUIRES log esp_idf_lib_helpers driver)
+                       REQUIRES log esp_idf_lib_helpers driver esp_timer)

--- a/components/sht3x/CMakeLists.txt
+++ b/components/sht3x/CMakeLists.txt
@@ -1,5 +1,11 @@
+if(${IDF_VERSION_MAJOR} STREQUAL 4 AND ${IDF_VERSION_MINOR} STREQUAL 1 AND ${IDF_VERSION_PATCH} STREQUAL 3)
+	set(req i2cdev log esp_idf_lib_helpers)
+else()
+	set(req i2cdev log esp_idf_lib_helpers esp_timer)
+endif()
+
 idf_component_register(
     SRCS sht3x.c
     INCLUDE_DIRS .
-    REQUIRES i2cdev log esp_idf_lib_helpers esp_timer
+    REQUIRES ${req} 
 )

--- a/components/sht3x/CMakeLists.txt
+++ b/components/sht3x/CMakeLists.txt
@@ -1,5 +1,5 @@
 idf_component_register(
     SRCS sht3x.c
     INCLUDE_DIRS .
-    REQUIRES i2cdev log esp_idf_lib_helpers
+    REQUIRES i2cdev log esp_idf_lib_helpers esp_timer
 )

--- a/components/sht4x/CMakeLists.txt
+++ b/components/sht4x/CMakeLists.txt
@@ -1,5 +1,5 @@
 idf_component_register(
     SRCS sht4x.c
     INCLUDE_DIRS .
-    REQUIRES i2cdev log esp_idf_lib_helpers
+    REQUIRES i2cdev log esp_idf_lib_helpers esp_timer
 )

--- a/components/sht4x/CMakeLists.txt
+++ b/components/sht4x/CMakeLists.txt
@@ -1,5 +1,11 @@
+if(${IDF_VERSION_MAJOR} STREQUAL 4 AND ${IDF_VERSION_MINOR} STREQUAL 1 AND ${IDF_VERSION_PATCH} STREQUAL 3)
+	set(req i2cdev log esp_idf_lib_helpers)
+else()
+	set(req i2cdev log esp_idf_lib_helpers esp_timer)
+endif()
+
 idf_component_register(
     SRCS sht4x.c
     INCLUDE_DIRS .
-    REQUIRES i2cdev log esp_idf_lib_helpers esp_timer
+    REQUIRES ${req}
 )

--- a/components/ultrasonic/CMakeLists.txt
+++ b/components/ultrasonic/CMakeLists.txt
@@ -1,5 +1,7 @@
 if(${IDF_TARGET} STREQUAL esp8266)
     set(req esp8266 freertos esp_idf_lib_helpers esp_timer)
+elseif(${IDF_VERSION_MAJOR} STREQUAL 4 AND ${IDF_VERSION_MINOR} STREQUAL 1 AND ${IDF_VERSION_PATCH} STREQUAL 3)
+    set(req driver freertos esp_idf_lib_helpers)
 else()
     set(req driver freertos esp_idf_lib_helpers esp_timer)
 endif()

--- a/components/ultrasonic/CMakeLists.txt
+++ b/components/ultrasonic/CMakeLists.txt
@@ -1,7 +1,7 @@
 if(${IDF_TARGET} STREQUAL esp8266)
-    set(req esp8266 freertos esp_idf_lib_helpers)
+    set(req esp8266 freertos esp_idf_lib_helpers esp_timer)
 else()
-    set(req driver freertos esp_idf_lib_helpers)
+    set(req driver freertos esp_idf_lib_helpers esp_timer)
 endif()
 
 idf_component_register(

--- a/components/wiegand/CMakeLists.txt
+++ b/components/wiegand/CMakeLists.txt
@@ -1,7 +1,7 @@
 if(${IDF_TARGET} STREQUAL esp8266)
-    set(req esp8266 log esp_idf_lib_helpers)
+    set(req esp8266 log esp_idf_lib_helpers esp_timer)
 else()
-    set(req driver log esp_idf_lib_helpers)
+    set(req driver log esp_idf_lib_helpers esp_timer)
 endif()
 
 idf_component_register(

--- a/components/wiegand/CMakeLists.txt
+++ b/components/wiegand/CMakeLists.txt
@@ -1,5 +1,7 @@
 if(${IDF_TARGET} STREQUAL esp8266)
     set(req esp8266 log esp_idf_lib_helpers esp_timer)
+elseif(${IDF_VERSION_MAJOR} STREQUAL 4 AND ${IDF_VERSION_MINOR} STREQUAL 1 AND ${IDF_VERSION_PATCH} STREQUAL 3)
+    set(req driver log esp_idf_lib_helpers)
 else()
     set(req driver log esp_idf_lib_helpers esp_timer)
 endif()

